src/hal/pwm.rs

```diff
@@ -0,0 +1,59 @@
+// Zinc, the bare metal stack for rust.
+// Copyright 2015 Paul Osborne <osbpau@gmail.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Common definitions for all PWM on all MCUs
+
+/// Trait for any Puluse Width Modulated output
+///
+/// This interface is inspired by the mbed `PWMOut` interface.
+///
+/// Note that on some MCUs, the period may be dictated by
+/// a timer shared by several differents PWMs.  It is not
+/// guaranteed that the period and pulsewidth specified
+/// will remain identical if modified elsewhere.  A proper
+/// implementaiton will still seek to maintain a similar
+/// duty cycle in the case of a period change.
+pub trait PWMOutput {
+  /// set the period in microseconds
+  fn set_period_us(&mut self, period_us: u32);
```

Comment:
Yeah, my interface is based on the mbed interface which I assumed was at least a reasonable starting point: https://developer.mbed.org/users/mbed_official/code/mbed/docs/cbbeb26dbd92//classmbed_1_1PwmOut.html